File: 백준/Silver/18111. 마인크래프트/마인크래프트.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);
        int b = Integer.parseInt(input[2]);

        int[] heights = new int[257];
        int mapSize = n * m;
        int blocks = 0;
        int minHeight = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            input = br.readLine().split(" ");
            for (int j = 0; j < m; j++) {
                int h = Integer.parseInt(input[j]);
                heights[h]++;
                blocks += h;
                minHeight = Math.min(h, minHeight);
            }
        }

        int maxHeight = b + blocks < (mapSize * 256) ? (b + blocks) / mapSize : 256;
        int t = (blocks - (minHeight * mapSize)) * 2;

        int time = t;
        int height = minHeight;
        blocks = 0;

        for (int i = minHeight + 1; i <= maxHeight; i++) {
            blocks += heights[i - 1];
            t += blocks - ((mapSize - blocks) * 2);

            if (time >= t) {
                time = t;
                height = i;
            }
        }

        System.out.println(time + " " + height);
    }
}

File: 백준/Silver/18111. 마인크래프트/마인크래프트.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

const [n, m, b] = input.shift().split(" ").map(Number);
let heights = Array.from({ length: 257 }).fill(0);
let mapSize = n * m;
let blocks = 0;
let minHeight = Number.MAX_SAFE_INTEGER;

input.map((arr) =>
  arr.split(" ").map((h) => {
    heights[Number(h)]++;
    blocks += Number(h);
    minHeight = Math.min(Number(h), minHeight);
  })
);

let maxHeight =
  b + blocks < mapSize * 256 ? Math.floor((b + blocks) / mapSize) : 256;
let t = (blocks - minHeight * mapSize) * 2;

let time = t;
let height = minHeight;
blocks = 0;

for (let i = minHeight + 1; i <= maxHeight; i++) {
  blocks += heights[i - 1];
  t += blocks - (mapSize - blocks) * 2;

  if (time >= t) {
    time = t;
    height = i;
  }
}

console.log(time, height);
